openpype/hosts/substancepainter/plugins/publish/collect_textureset_images.py

```diff
@@ -0,0 +1,177 @@
+import os
+import copy
+import pyblish.api
+
+from openpype.pipeline import publish
+
+import substance_painter.textureset
+from openpype.hosts.substancepainter.api.lib import (
+    get_parsed_export_maps,
+    strip_template
+)
+from openpype.pipeline.create import get_subset_name
+from openpype.client import get_asset_by_name
+
+
+class CollectTextureSet(pyblish.api.InstancePlugin):
+    """Extract Textures using an output template config"""
+    # TODO: Production-test usage of color spaces
+    # TODO: Detect what source data channels end up in each file
+
+    label = "Collect Texture Set images"
+    hosts = ['substancepainter']
+    families = ["textureSet"]
+    order = pyblish.api.CollectorOrder
+
+    def process(self, instance):
+
+        config = self.get_export_config(instance)
+        asset_doc = get_asset_by_name(
+            project_name=instance.context.data["projectName"],
+            asset_name=instance.data["asset"]
+        )
+
+        instance.data["exportConfig"] = config
+        maps = get_parsed_export_maps(config)
+
+        # Let's break the instance into multiple instances to integrate
+        # a subset per generated texture or texture UDIM sequence
+        for (texture_set_name, stack_name), template_maps in maps.items():
+            self.log.info(f"Processing {texture_set_name}/{stack_name}")
+            for template, outputs in template_maps.items():
+                self.log.info(f"Processing {template}")
+                self.create_image_instance(instance, template, outputs,
+                                           asset_doc=asset_doc)
+
+    def create_image_instance(self, instance, template, outputs,
+                              asset_doc):
+        """Create a new instance per image or UDIM sequence.
+
+        The new instances will be of family `image`.
+
+        """
+
+        context = instance.context
+        first_filepath = outputs[0]["filepath"]
+        fnames = [os.path.basename(output["filepath"]) for output in outputs]
+        ext = os.path.splitext(first_filepath)[1]
+        assert ext.lstrip('.'), f"No extension: {ext}"
```

Comment:
```suggestion
        assert ext.lstrip("."), f"No extension: {ext}"
```

But I am not sure about this assert, if there is a changes there might not be extension, shouldn't we throw some Exception instead?